compiler/rustc_lint/src/map_unit_fn.rs

```diff
@@ -80,11 +83,14 @@ impl<'tcx> LateLintPass<'tcx> for MapUnitFn {
                                 MAP_UNIT_FN,
                                 span,
                                 MappingToUnit {
-                                    function_label: cx.tcx.span_of_impl(*id).unwrap(),
                                     argument_label: args[0].span,
                                     map_label: arg_ty.default_span(cx.tcx),
                                     suggestion: path.ident.span,
                                     replace: "for_each".to_string(),
+                                    func_label: MappingToUnitFuncLabel {
+                                        span: cx.tcx.span_of_impl(*id).map_or(None, |s| Some(s)),
```

Comment:
```suggestion
                                        span: cx.tcx.span_of_impl(*id).ok(),
```

But I would prefer if you made `MappingToUnitFuncLabel` have two variants